File: 3лаб_16варКЗП/лаб3/Lab3/src/KI/Milian/Lab3/Battery.java
package KI.Milian.Lab3;

/**
 * Клас, що описує акумулятор для аудіоплеєра.
 */
public class Battery {
    private int capacity;  // Ємність батареї в мАг (мілліампер-години)

    public Battery(int capacity) {
        this.capacity = capacity;
    }

    public int getCapacity() {
        return capacity;
    }

    public void setCapacity(int capacity) {
        this.capacity = capacity;
    }

    /**
     * Зменшує заряд батареї на вказану кількість мАг.
     * Якщо заряд стає меншим за 0, встановлюється на 0.
     */
    public void drainBattery(int amount) {
        capacity -= amount;
        if (capacity < 0) {
            capacity = 0;
        }
    }

    @Override
    public String toString() {
        return "Battery{capacity=" + capacity + "mAh}";
    }
}

File: 9лаб_16варКЗП/лаб9/Battery.py
# Клас Battery представляє батарею з певною ємністю, що дозволяє керувати її зарядом.
class Battery:
    def __init__(self, capacity):
        # Ініціалізує батарею з заданою ємністю.
        self.capacity = capacity

    # Метод для отримання поточної ємності батареї.
    def get_capacity(self):
        return self.capacity

    # Метод для встановлення нової ємності батареї.
    def set_capacity(self, capacity):
        self.capacity = capacity

    # Метод для розряду батареї на вказану величину.
    def drain_battery(self, amount):
        self.capacity -= amount  # Зменшує ємність батареї на вказану величину.
        if self.capacity < 0:
            self.capacity = 0  # Якщо ємність менша за 0, встановлює її в 0.

    # Метод для формування рядкового представлення об'єкта Battery.
    def __str__(self):
        # Повертає інформацію про поточну ємність батареї в мА·год.
        return f"Battery{{capacity={self.capacity}mAh}}"
